### ISSUE TITLE
Better messaging when failing by type 

### ISSUE THREAD

**Issue by duvanmonsa** [cypress-io/cypress#2050]
### Current behavior:
When you try to validate values if the type is not the same,  the message is so confusing

<img width="391" alt="screen shot 2018-06-27 at 2 43 42 pm" src="https://user-images.githubusercontent.com/2709271/41996690-4d0603e8-7a1b-11e8-9949-dbb4f761b3b5.png">

### Desired behavior:
It should say that the types are different! like
### Steps to reproduce:
```js
const amount = 2000;
cy.get('.input-class')
      .type(amount)
      .should('have.value', amount);
```

### Versions
Cypress package version: 3.0.1
Cypress binary version: 3.0.1

**Comment by jennifer-shehane** [comment]
I thought we had fixed this before - and indeed we had fixed this within the Cypress Error printed below this message:

<img width="386" alt="screen shot 2018-06-27 at 5 41 35 pm" src="https://user-images.githubusercontent.com/1271364/42001249-7307b7ec-7a31-11e8-9856-40fc40c72ab5.png">

I feel like this message is printed from chai, but is editable. Here? @brian-mann? https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/chai_jquery.coffee#L116

**Comment by jennifer-shehane** [comment]
Also, some related conversation: https://github.com/cypress-io/cypress/issues/1360#issuecomment-368277204

**Comment by jennifer-shehane** [comment]
This was fixed in 3.7.0 as part of this PR https://github.com/cypress-io/cypress/pull/5619. Please update to the [current version of Cypress](https://on.cypress.io/changelog).

<img width="402" alt="Screen Shot 2020-01-07 at 3 03 12 PM" src="https://user-images.githubusercontent.com/1271364/71880524-f0829d00-315e-11ea-8c77-31c23af66763.png">
